distributed/http/tests/test_worker_http.py

```diff
@@ -35,3 +50,26 @@ def test_services(s, a, b):
     assert isinstance(c.services['http'], HTTPServer)
     assert c.service_ports['http'] == c.services['http'].port
     assert s.worker_services[c.address]['http'] == c.service_ports['http']
+
+
+@gen_cluster()
+def test_with_data(s, a, b):
+    e = Executor((s.ip, s.port), start=False)
+    yield tornado.gen.sleep(0.5)
+    yield e._start()
+    yield e._scatter([1])
+    yield tornado.gen.sleep(0.5)
+    client = AsyncHTTPClient()
+    d = yield e.scheduler.identity()
+    sch_hport = services['http']
+    (wip, wport) = list(e.workers)[0]
```

Comment:
Does `e.workers` do anything?  I recommend using `s.ncores` as the single point of truth on what workers are connected.  You could also use `a.address` or `b.address`.